pkg/commands/plan/plan_run.go

```diff
@@ -286,22 +305,22 @@ func (c *PlanRunCommand) updateResultCommentForActions(ctx context.Context, star
 			fmt.Fprintf(&comment, "%s 🟩 Successful for dir: `%s` %s", CommentPrefix, c.childPath, c.gitHubLogURL)
 		}
 		if result.commentDetails != "" {
-			fmt.Fprintf(&comment, "\n\n<details>\n<summary>Details</summary>\n\n```diff\n\n%s\n```\n</details>", result.commentDetails)
+			// Ensure the comment is not over GitHub's limit. We need to account for the surrounding characters we will
+			// be adding in addition to the length of result.commentDetails.
+			fmtString := "\n\n<details>\n<summary>Details</summary>\n\n```diff\n\n%s\n```\n</details>"
+			truncationMsg := []rune("...\n\nMessage has been truncated. See workflow logs to view the full message.")
+			cappedLength := gitHubMaxCommentLength - len(truncationMsg) - len([]rune(comment.String())) - len([]rune(fmtString)) + 2
+			if len([]rune(result.commentDetails)) > cappedLength {
+				runes := []rune(result.commentDetails)[:cappedLength]
+				runes = append(runes, truncationMsg...)
```

Comment:
Should we move the truncation message outside of the details so its obvious on the comment? Rather than the user having to go the entire way through the output only to realize its truncated?

Reply:
I think that is probably a better user experience. I've moved the truncation message outside the comment details section of the comment.